src/utils/cid/mod.rs

```diff
@@ -89,15 +101,7 @@ impl<'de> Deserialize<'de> for SmallCid {
 
 impl From<Cid> for SmallCid {
     fn from(cid: Cid) -> Self {
-        if cid.version() == Version::V1 && cid.codec() == DAG_CBOR {
-            if let Ok(small_hash) = cid.hash().resize() {
-                let (code, bytes, size) = small_hash.into_inner();
-                if code == u64::from(Code::Blake2b256) && size as usize == BLAKE2B256_SIZE {
-                    return SmallCid(SmallCidInner::V1DagCborBlake2b(bytes));
-                }
-            }
-        }
-        SmallCid(SmallCidInner::Generic(Box::new(cid)))
+        SmallCid::minimal_cid(SmallCid(SmallCidInner::Generic(Box::new(cid))))
```

Comment:
This is a bit of a smell - this is what prompts the refactoring of the above function

Reply:
Corrected.